template/checkNothing/checkNothing.proto

```diff
@@ -0,0 +1,30 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package checkNothing;
+
+import "pkg/adapter/template/TemplateExtensions.proto";
+
+option (istio.mixer.v1.config.template.template_variety) = TEMPLATE_VARIETY_CHECK;
```

Comment:
I don't remember this ask of moving annotations inside message (just created a new issue for this https://github.com/istio/mixer/issues/1040). However, it should be a trivial change. we can make this change.. it is a find and replace issue on the template itself (and codegen implementation) and nothing else would change.. I would prefer to do this as a separate refactor step (I can get to it within 2-3 days once I finish the task in hand) and bulk fix the existing template 

Regarding renaming of the Template message to a desired name message. I think calling it Template is very explicit and come handy once we start supporting sub messages and all the sub messages will be also defined inside the same proto file. It will help in finding which is the uber container root message. If we replace Template with any user defiend name, then the presence of template_variety will dictate the main root message. I personally like this explicit 'Template' declaration schema.. what do others think @ZackButcher @mandarjog @douglas-reid  ? Note if we rename 'Template' to 'ListEntry' then we also ditch the template_name option and simply just reuse the message name.